src/playback-watcher.js

```diff
@@ -204,16 +204,18 @@ export default class PlaybackWatcher {
 
     this[`${type}StalledDownloads_`]++;
 
-    this.logger_(`found stalled download #${this[`${type}StalledDownloads_`]} for ${type} loader`);
-    // We will technically get past this on the fourth bad append
-    // rather than the third. As the first will almost always cause
-    // buffered to change which means that StalledDownloads_ will
-    // not be incremented
-    if (this[`${type}StalledDownloads_`] < 3) {
+    this.logger_(`found #${this[`${type}StalledDownloads_`]} ${type} appends that did not increase buffer (possible stalled download)`, {
+      playlistId: loader.playlist_ && loader.playlist_.id,
+      buffered: Ranges.timeRangesToArray(buffered)
+
+    });
+
+    // after 5 possibly stalled appends with no reset, exclude
+    if (this[`${type}StalledDownloads_`] < 5) {
```

Comment:
Wait until 5 stalled downloads.

Reply:
Let's use a larger value here. Would 10 be ok? Maybe 15 or even 20?